src/measure_theory/integral/layercake.lean

```diff
@@ -226,3 +237,116 @@ end
 end measure_theory
 
 end layercake
+
+section layercake_lt
+
+open measure_theory
+
+variables {α : Type*} [measurable_space α] (μ : measure α)
+variables {β : Type*} [measurable_space β] [measurable_singleton_class β]
+
+namespace measure
+
+lemma meas_le_ne_meas_lt_subset_meas_pos {R : Type*} [linear_order R]
+  [measurable_space R] [measurable_singleton_class R] {g : α → R} (g_mble : measurable g) :
+  {t : R | μ {a : α | t ≤ g a} ≠ μ {a : α | t < g a}} ⊆ {t : R | 0 < μ {a : α | g a = t}} :=
+begin
+  intro t,
```

Comment:
Is there a reason not to use the simpler
```suggestion
  [measurable_space R] [measurable_singleton_class R] {g : α → R} (g_mble : measurable g)
  {t} :
  μ {a : α | t ≤ g a} ≠ μ {a : α | t < g a}→  0 < μ {a : α | g a = t} :=
begin
```

Reply:
No particular reason except I thought in terms of subsets where I used this... I will try replacing with the simpler form you suggested, and if this (hopefully definitionally equivalent) form still works smoothly in the main proof, I will keep it. Thank you!